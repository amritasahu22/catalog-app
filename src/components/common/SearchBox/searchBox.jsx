import React from "react";
import PropTypes from "prop-types";

const SearchBox = (props) => {
	const { value, onChange } = props;
	return (
		<input
			type="text"
			name="query"
			className="form-control my-3"
			placeholder="Search..."
			value={value}
			onChange={(e) => onChange(e.currentTarget.value)}
		/>
	);
};

SearchBox.propTypes = {
	value: PropTypes.object,
	onChange: PropTypes.func.isRequired,
};
export default SearchBox;
