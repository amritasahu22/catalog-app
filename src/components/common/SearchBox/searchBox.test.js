import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import SearchBox from "./SearchBox";

describe("SearchBox", () => {
	test("should render properly with props", () => {
		const value = "2xu";
		const component = shallow(<SearchBox value={value} onChange={() => {}} />);
		expect(toJson(component)).toMatchSnapshot();
	});
});
