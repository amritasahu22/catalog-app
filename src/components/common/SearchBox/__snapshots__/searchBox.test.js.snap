// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBox should render properly with props 1`] = `
<input
  className="form-control mx-3"
  name="query"
  onChange={[Function]}
  placeholder="Search..."
  type="text"
  value="2xu"
/>
`;
